src/Configuration.cpp

```diff
@@ -233,18 +233,29 @@ Configuration::makeDispatcher(std::shared_ptr<TimeWarpCommunicationManager> comm
 
         std::unique_ptr<TimeWarpEventSet> event_set = make_unique<TimeWarpEventSet>();
 
+// If we only want one thread per queue, then we tie the amount of worker treads to the amount of schedule queues
+// Number of worker threads is defined after the number of schedule queues if defined below
+#if !defined (ONE_THREAD_PER_LTSF)
         // WORKER THREADS
         int num_worker_threads = (*root_)["time-warp"]["worker-threads"].asInt();
         if (!checkTimeWarpConfigs(num_worker_threads, all_config_ids, comm_manager)) {
             invalid_string += std::string("\tNumber of worker threads\n");
         }
+#endif
 
         // SCHEDULE QUEUES
         int num_schedulers = (*root_)["time-warp"]["scheduler-count"].asInt();
         if (!checkTimeWarpConfigs(num_schedulers, all_config_ids, comm_manager)) {
             invalid_string += std::string("\tNumber of schedule queues\n");
         }
 
+// If we disable schedule queue locks, then we need to make sure that there are 
+// exactly one worker thread per schedule queue
+#if defined (ONE_THREAD_PER_LTSF)
+        int num_worker_threads = num_schedulers;
+        assert(num_worker_threads%num_schedulers == 0);
```

Comment:
get rid of this assert here. It is not needed since you're assigning the values in the previous line of code. Just place an assert after line 247 : assert(num_schedulers > 0)